treeherder/intermittents_commenter/commenter.py

```diff
@@ -70,6 +70,8 @@ def generate_bug_changes(self, startday, endday, alt_startday, alt_endday):
             # recommend disabling when more than 150 failures tracked over 21 days
             if alt_bug_stats[bug_id]['total'] >= 150:
                 bug_info, whiteboard = self.check_bug_info(bug_info, bug_id)
+                if bug_info is None:
```

Comment:
I think `check_bug_info()` is a bit strange at the moment. It's acting a bit like a cache (except the cached value has to be passed in?!) and then it also extracts the whiteboard value, which seems like something that should be done by the caller, and not inside a function. That said, perhaps it's just another thing that needs to be added to the "things to refactor later" list - since this whole class could do with a bit of TLC once the switchover has occurred.

Anyway, following the call chain, I see that the reason `bug_info` can be None, is that if the request to Bugzilla's API fails, then `fetch_bug_details()` returns `None`. This wasn't immediately obvious from this particular line, so I think a code comment is needed here and for the other instances below.

Reply:
Yeah happy to wait until switchover :-)